templates/flavors/default/patches/control-plane.yaml

```diff
@@ -0,0 +1,15 @@
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureCluster
+metadata:
+  name: ${CLUSTER_NAME}
+spec:
+  networkSpec:
+    apiServerLB:
+      # Hack: We pre-create this public IP and the DNS name to use it in the
```

Comment:
We can get rid of this comment, as (arguably) this is a normal configuration and not a hack, and the security rules referred to are not universal across Azure.

Reply:
Another solution would be to update the IP address that is passed to CAPI kubeadm.  The approach taken here has the advantage of not needing to have the management cluster peered to the worker node vnet. I guess scenarios where you wanted it fully private you could go with a private cluster and bolt on a public IP address with ASO, if desired, later once https://github.com/Azure/azure-service-operator/issues/4368 is resolved.

Anyways keeping some form of comment here is probably useful for our future selves.

Reply:
I like the idea of a comment that clarifies any housekeeping to ensure that nodes are able to connect to the apiserver locally (whether or not there is a private or public apiserver endpoint), but I don't think we want to reference a "security policy", which is not general for Azure, and is subject to change.

Reply:
> Another solution would be to update the IP address that is passed to CAPI kubeadm

Is that IP Address present in `/etc/kubernetes/kubelet.conf` as referenced in Jack's proposal here https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5209#discussion_r1821104012 ?

Reply:
> I like the idea of a comment that clarifies any housekeeping to ensure that nodes are able to connect to the apiserver locally (whether or not there is a private or public apiserver endpoint), but I don't think we want to reference a "security policy", which is not general for Azure, and is subject to change.

Addressed via 4996a360703222d42f30b2566deff68d68eb966d